packages/graphql/lib/src/core/observable_query.dart

```diff
@@ -123,11 +123,15 @@ class ObservableQuery {
         queryManager.fetchQueryAsMultiSourceResult(queryId, options);
     latestResult ??= allResults.eagerResult;
 
-    // if onData callbacks have been registered,
-    // they are waited on by default
-    lifecycle = _onDataSubscriptions.isNotEmpty
-        ? QueryLifecycle.SIDE_EFFECTS_PENDING
-        : QueryLifecycle.PENDING;
+    if (allResults.networkResult != null) {
+      // if onData callbacks have been registered,
+      // they are waited on by default
+      lifecycle = _onDataSubscriptions.isNotEmpty
+          ? QueryLifecycle.SIDE_EFFECTS_PENDING
+          : QueryLifecycle.PENDING;
+    } else if (allResults.eagerResult != null) {
```

Comment:
@Sh1d0w do you think it makes sense here, what it did is to satisfy [here](https://github.com/zino-app/graphql-flutter/blob/42d010c177f7fa752b9a0b5ca57548dd1cd3ad8f/packages/graphql/lib/src/core/observable_query.dart#L62) to make `cacheFirst` can be [refetch](https://github.com/zino-app/graphql-flutter/blob/42d010c177f7fa752b9a0b5ca57548dd1cd3ad8f/packages/graphql/lib/src/core/observable_query.dart#L81), I'm not very sure it won't have the side effect.

Reply:
I don't think this change is necessary. Side effects are only possible with mutations, but refetch is only possible with queries, so there shouldn't be any kind of collision.